### DISCUSSION TITLE
Load SSL Certificates from Memory Buffer.

### DISCUSSION THREAD

**Discussion by FBaumeister** [edenhill/librdkafka#3680] [Q&A]
Hi,
when I set up my Producer via the config using ssl.ca.location everything works fine.
Now I want to set it up via **set_ssl_cert** but I get strange behaviour as config variables for ssl.ca.pem and ssl.certificate.pem and ssl.key.location do all seem to contain the same value after manipulating some entries with set_ssl_cert.
Now on top of that I need to use DER Format with set_ssl_cert but I really cannot figure out how or better which config variables I have to provide when providing the certificates from a memory buffer.
Can somebody help me?
Thank you.

**Comment by edenhill** [comment]
I believe something like `rd_kafka_conf_set_ssl_cert(conf, RD_KAFKA_CERT_CA, RD_KAFKA_CERT_ENC_PEM, your_pem_blob, your_pem_blob_size, errstr, sizeof(errstr));`.

But you could just as well just pass the PEM string directly to the `ssl.ca.pem` config property.
